### ISSUE TITLE
Connect dockerhub to build images

### ISSUE THREAD

**Issue by cgetzen** [tmate-io/tmate#175]
There's already a dockerfile. This will allow people to do something like this in their own dockerfiles:
```
FROM tmate-io/tmate AS tmate
FROM actual_base_image AS base
COPY --from=tmate /build/tmate /bin/tmate
...
Rest of dockerfile
````

**Comment by nviennot** [comment]
Hello,

What would be the benefit compared to using the following?

```
RUN wget -O - https://github.com/tmate-io/tmate/releases/download/2.4.0/tmate-2.4.0-static-linux-amd64.tar.xz | tar --strip-components 1 -xJf -
```

**Comment by cgetzen** [comment]
You're right, there's not a huge difference. However, by publishing the docker image, we wouldn't have to install wget before pulling the tarball. A tmate image could also be used as remote busybox.

**Comment by nviennot** [comment]
wget is part of busybox though.

I've added a tmate image: https://hub.docker.com/repository/docker/tmate/tmate
Let me know if this is acceptable.

**Comment by pataquets** [comment]
I'd say there is at least one difference: no source builds.
Also, Docker Hub's Automated Builds are now a paid service. I've replaced it with Github Actions, if you're interested.
I can help, if there is interest. See my somewhat related [tmux Docker builder](https://github.com/pataquets/tmux-builder/).
